### ISSUE TITLE
Sending Status.Failure as a response to an Ask made from a different cluster node results in path resolution failure

### ISSUE THREAD

**Issue by ondrejpialek** [akkadotnet/akka.net#6067]
**Version Information**

Using Akka, Cluster Sharding, DData and others, all version `1.4.40`.

**Description**

We have recently refactored our code to return `Status.Failure` instead of our own generic response wrappers that carry success / failure info. The benefit we were after was to have `await Ask` to throw, which it should if it receives a `Status.Failure`.

When the Ask targets a remote actor and `Status.Failure` is sent from the other cluster node, it is not delivered successfully, but rather a path resolution failure is logged and the `Status.Failure` message is sent to dead letters.

This looks very similar to https://github.com/akkadotnet/akka.net/issues/4384 but might be a special case of that related to sending `Status.Failure`. Aside from the special handling it gets, another interesting aspect of that class is that it sends an Exception, which notoriously has problems with serialization, but I am unsure if this is even relevant for this issue.

**Screenshots**
![image](https://user-images.githubusercontent.com/580850/185808627-a2364db9-8ad2-4e7a-b965-5a09bbccb232.png)

**Environment**
Windows, .NET 6

**Relevant excerpts from logs**

- `Resolve of path sequence [/temp/w] failed` from `akka://SKChase-Pepper/system/endpointManager/reliableEndpointWriter-akka.tcp%3A%2F%2FSKChase-Pepper%40moon%3A30030-3/endpointWriter/endpointReader-akka.tcp%3A%2F%2FSKChase-Pepper%40moon%3A30030-1`

- `Message [Failure] from [akka.tcp://SKChase-Pepper@moon:30030/user/OrderDomainService/GetOrder-3fe063c1-a1da-4204-80c8-e8d5bad410ed#1224953750] to [akka://SKChase-Pepper/temp/w] was not delivered. [3] dead letters encountered. If this is not an expected behavior then [akka://SKChase-Pepper/temp/w] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.` from `akka://SKChase-Pepper/system/deadLetterListener`

**Comment by Aaronontheweb** [comment]
@Arkatufus this sounds like something we could reproduce

**Comment by Arkatufus** [comment]
@ondrejpialek Are you using JSON or Hyperion serializer?

**Comment by ondrejpialek** [comment]
@Arkatufus I believe it's Hyperion for remoting. We use protobuf for persistance but that should not be relevant here I suppose.

**Comment by Arkatufus** [comment]
@ondrejpialek 
I couldn't reproduce the issue using both direct remote deployment and actor selector.
How do you obtain the remote `IActorRef` reference you're `Ask`-ing to?
Are you sure that the `Ask` operation didn't time-out?

**Comment by Arkatufus** [comment]
@ondrejpialek 
The only way I could reproduce your logs is to timeout the `Ask` operation.
When the `Ask` operation timed-out, the temporary actor is disposed, which explains why the `ActorSystem` failed to resolve the path and why the message is being sent to `DeadLetters`

**Comment by ondrejpialek** [comment]
Thanks @Arkatufus, I will try to narrow it down on our end to see if I can come up with a more specific example.

We do use timeouts, but for this we are passing a generic timeout of 30 minutes. From the logs I can see that the reply is being sent in a couple hundered miliseconds after the Ask.

Something that is perhaps not so standard here is that we ask via the pub sub mediator, using the `Send` message and passing in `localAffinity` as `true`. Now the odd thing is that it is still processed on a different node, so I will experiment with this if it has any effect.

**Comment by ondrejpialek** [comment]
@Arkatufus I have identified the and I can confirm this is NOT an Akka bug, apologies for wasting your time on this!

It turns out that we have stopped a parent actor of the one that Asked, and when I was looking at the code I wrongly assumed it was the Ask made by the parent, when the failing Ask was made by the child.

I missed this party because Ask "actors" live under the `temp` path so it's not possible to tell where they were created just by looking at their path, but in any case I should have had better logging to lead me to the right place in the first place.

Apologies again and thanks for a stellar job you'd doing here on Github.
